java/core/src/java/org/apache/orc/filter/VectorFilter.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.orc.filter;
+
+import org.apache.orc.OrcFilterContext;
+import org.apache.orc.filter.impl.Selected;
+
+/**
+ * A filter that operates on the supplied
+ * {@link org.apache.hadoop.hive.ql.exec.vector.VectorizedRowBatch} and updates the selections.
+ *
+ * This is the interface that is the basis of both the leaf filters such as Equals, In and logical
+ * filters such as And, Or and Not
+ */
+public interface VectorFilter {
+
+  /**
+   * Filter the vectorized row batch that is wrapped into the FilterContext.
+   * @param fc     The filter context that contains the VectorizedRowBatch
+   * @param bound  The bound of the scan
+   * @param selIn  The current selection
+   * @param selOut The result selection
+   */
+  void filter(OrcFilterContext fc, Selected bound, Selected selIn, Selected selOut);
```

Comment:
I'd propose that we should join bound and selIn to be the rows that should be checked. The documentation should make it clear that bound should not be modified. Furthermore, we document that items in selOut that are not in bound must be retained. The selOut vector must be sorted.

Reply:
Makes sense.

* Removed selIn
* Updated the documentation